shell/debug.sh

```diff
@@ -83,11 +85,37 @@ if [[ $HEADLESS == "true" ]]; then
   )
 fi
 
-# When not running in a container, we can start without logging.
-# Otherwise, we need to log to a file so that the output can be
-# processed by devspace.
-if [[ $IN_CONTAINER == "false" ]]; then
+function ctrl_c_trap() {
+  echo "killing delve with PID: $DLV_PID"
+  kill "$DLV_PID"
+  exit 0
+}
+trap ctrl_c_trap SIGINT
+
+if [[ $HEADLESS == "false" ]]; then
   exec "${delve[@]}"
 else
-  exec "${delve[@]}" | tee -ai "$DEV_CONTAINER_LOGFILE"
+
+  # Start headless delve in the background so we can kill it with ctrl-c.
+
+  if [[ $IN_CONTAINER == "false" ]]; then
+    # no need to log outside of a container
+    "${delve[@]}" &
+  else
+    # We only need to start logging if we are running in a
+    # a container so the output can be processed by devspace.
+    echo -e "\n\n\n\n\n\n\n\n" >>"$DEV_CONTAINER_LOGFILE"
+    "${delve[@]}" >>"$DEV_CONTAINER_LOGFILE" 2>&1 &
+    IS_LOGGING="true"
+  fi
+
+  DLV_PID=$!
+  echo "delve pid is: $DLV_PID"
+
+  if [[ $IS_LOGGING == "true" ]]; then
```

Comment:
From Jareds comment I think we want to check if `HEADLESS=false && IN_CONTAINER` before tailing?

Reply:
True. Fixed that